src/request-api.js

```diff
@@ -90,6 +90,10 @@ function requestAPI (path, args, opts, files, buffer, cb) {
     })
   })
 
+  req.on('error', function (err) {
+    return cb(err, null)
+  })
```

Comment:
ahh yep. :+1: 

is it guaranteed that `http.request` only calls _either_ the callback (i.e. emits `response`) or emits `error` ? (i believe so, but asking others). cc @diasdavid